mrmap/registry/api/serializers/mapcontext.py

```diff
@@ -1,69 +1,175 @@
+            'offerings',  # 7.1.2.13 offering
+            # TODO 'active' # 7.1.2.14 active
+            # TODO 'categories' # 7.1.2.16 keyword
+            # TODO 'minScaleDenominator' # 7.1.2.17 minScaleDenominator
+            # TODO 'maxScaleDenominator' # 7.1.2.18 maxScaleDenominator
+            'folder',  # 7.1.2.19 folder
+            # TODO '*', # 7.1.2.20 extension
         ]
 
     @staticmethod
-    def get_max_extent(obj):
-        pass
-        # TODO
+    def __get_wms_offering(wms_layer):
+        return {
+            'code': 'http://www.opengis.net/spec/owc-geojson/1.0/req/wms',
+            'operations': [{
+                "code": "GetCapabilities",
+                "method": "GET",
+                "type": "application/xml",
+                "href": f"{wms_layer.service.url}service=WMS&request=GetCapabilities"
+            }, {
+                "code": "GetMap",
+                "method": "GET",
+                "type": "image/jpeg",  # TODO
+                "href": f"{wms_layer.service.url}service=WMS&version=1.3.0&request=GetMap&layer={wms_layer.identifier}"
+                # TODO full request url
+            }]
+        }
+
+    @staticmethod
+    def get_offerings(obj):
+        offerings = [MapContextLayerPropertiesSerializer.__get_wms_offering(obj.layer)]
+        return offerings
 
     @staticmethod
-    def get_context_layer_operations(obj):
-        map_context_layer = MapContextLayer.objects.filter(map_context_id=obj.id)
-        layer = map_context_layer.get().layer
-        service = layer.service
-        operations = OperationUrl.objects.filter(service__id=service.id)
-        return OperationsUrlSerializer(operations, many=True).data
+    def get_folder(obj):
+        folder = None
+        while obj.parent and obj.parent.name != "/":
+            obj = obj.parent
+            if folder:
+                folder = obj.name + "/" + folder
+            else:
+                folder = obj.name
+        return folder
 
 
-class MapContextSerializer(ModelSerializer):
-    type = CharField(default='FeatureCollection')
+class MapContextLayerSerializer(NonNullModelSerializer):
+    type = CharField(default='Feature')
     id = SerializerMethodField()
-    # properties = MapContextPropertiesSerializer()
-    map_context_layers = SerializerMethodField()
+    properties = MapContextLayerPropertiesSerializer(source='*')
 
     class Meta:
-        model = MapContext
+        model = MapContextLayer
         fields = [
             'type',
-            'id',
-            # 'properties',
-            'map_context_layers'
+            'id',  # 7.1.2.1 id
+            'properties',
+            # TODO 'geometry'  # 7.1.2.8 geospatialExtent
+        ]
+
+    def get_id(self, obj):
+        return self.context['request'].build_absolute_uri(
+            reverse('api:mapcontext-detail', args=[obj.map_context_id]) + "layer/" + str(obj.id) + "/")
```

Comment:
like the current `GenericModelMixin` we should think about getting the rest api endpoint from the model obj it self.